### ISSUE TITLE
Swing Update manager - NullPointerException

### ISSUE THREAD

**Issue by TheAifam5** [NationalSecurityAgency/ghidra#1754]
**Describe the bug**
Error occurred after second analysis pass on libstdc++.so.6

**To Reproduce**
1. Analysis -> Auto Analyze -> Analyze
2. Wait until completes
3. Analysis -> Auto Analyze -> Analyze
4. Wait until completes

**Attachments**
```
2020-04-12	19:05:03	ERROR	(SwingUpdateManager) Unexpected Exception: Unexpected exception in Swing Update Manager java.lang.NullPointerException
			at ghidra.app.plugin.core.analysis.AutoAnalysisPlugin$OneShotAnalyzerAction.isEnabledForContext(AutoAnalysisPlugin.java:374)
			at docking.DockingActionProxy.isEnabledForContext(DockingActionProxy.java:189)
			at docking.WindowActionManager.processContextChanged(WindowActionManager.java:130)
			at docking.WindowActionManager.lambda$new$0(WindowActionManager.java:36)
			at ghidra.util.task.SwingUpdateManager.doWork(SwingUpdateManager.java:320)
			at ghidra.util.task.SwingUpdateManager.checkForWork(SwingUpdateManager.java:260)
			at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
			at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
			at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
			at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
			at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
			at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
			at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
			at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
			at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
			at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
			at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
			at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
			at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
			
2020-04-12	19:05:07	ERROR	(SwingUpdateManager) Unexpected Exception: Unexpected exception in Swing Update Manager java.lang.NullPointerException
			at ghidra.app.plugin.core.analysis.AutoAnalysisPlugin$OneShotAnalyzerAction.isEnabledForContext(AutoAnalysisPlugin.java:374)
			at docking.DockingActionProxy.isEnabledForContext(DockingActionProxy.java:189)
			at docking.WindowActionManager.processContextChanged(WindowActionManager.java:130)
			at docking.WindowActionManager.lambda$new$0(WindowActionManager.java:36)
			at ghidra.util.task.SwingUpdateManager.doWork(SwingUpdateManager.java:320)
			at ghidra.util.task.SwingUpdateManager.checkForWork(SwingUpdateManager.java:260)
			at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
			at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
			at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
			at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
			at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
			at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
			at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
			at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
			at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
			at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:117)
			at java.desktop/java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:190)
			at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:235)
			at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:233)
			at java.base/java.security.AccessController.doPrivileged(AccessController.java:312)
			at java.desktop/java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:233)
			at java.desktop/java.awt.Dialog.show(Dialog.java:1070)
			at java.desktop/java.awt.Component.show(Component.java:1716)
			at java.desktop/java.awt.Component.setVisible(Component.java:1663)
			at java.desktop/java.awt.Window.setVisible(Window.java:1021)
			at java.desktop/java.awt.Dialog.setVisible(Dialog.java:1005)
			at docking.DockingDialog.setVisible(DockingDialog.java:341)
			at docking.ErrLogDialog.repack(ErrLogDialog.java:230)
			at docking.ErrLogDialog.showDetails(ErrLogDialog.java:208)
			at docking.ErrLogDialog.lambda$buildMainPanel$1(ErrLogDialog.java:148)
			at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1967)
			at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2308)
			at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405)
			at java.desktop/javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262)
			at java.desktop/javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:279)
			at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:297)
			at java.desktop/java.awt.Component.processMouseEvent(Component.java:6636)
			at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3342)
			at java.desktop/java.awt.Component.processEvent(Component.java:6401)
			at java.desktop/java.awt.Container.processEvent(Container.java:2263)
			at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5012)
			at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2321)
			at java.desktop/java.awt.Component.dispatchEvent(Component.java:4844)
			at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4918)
			at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4547)
			at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4488)
			at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2307)
			at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2762)
			at java.desktop/java.awt.Component.dispatchEvent(Component.java:4844)
			at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:772)
			at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
			at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
			at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
			at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
			at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:95)
			at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:745)
			at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:743)
			at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
			at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
			at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:742)
			at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
			at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
			at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:117)
			at java.desktop/java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:190)
			at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:235)
			at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:233)
			at java.base/java.security.AccessController.doPrivileged(AccessController.java:312)
			at java.desktop/java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:233)
			at java.desktop/java.awt.Dialog.show(Dialog.java:1070)
			at java.desktop/java.awt.Component.show(Component.java:1716)
			at java.desktop/java.awt.Component.setVisible(Component.java:1663)
			at java.desktop/java.awt.Window.setVisible(Window.java:1021)
			at java.desktop/java.awt.Dialog.setVisible(Dialog.java:1005)
			at docking.DockingDialog.setVisible(DockingDialog.java:341)
			at docking.DockingWindowManager.lambda$doShowDialog$6(DockingWindowManager.java:1727)
			at ghidra.util.Swing.doRun(Swing.java:292)
			at ghidra.util.Swing.runNow(Swing.java:208)
			at ghidra.util.Swing.runNow(Swing.java:163)
			at docking.DockingWindowManager.doShowDialog(DockingWindowManager.java:1731)
			at docking.DockingWindowManager.showDialog(DockingWindowManager.java:1763)
			at docking.DockingErrorDisplay.doShowDialog(DockingErrorDisplay.java:103)
			at docking.DockingErrorDisplay.lambda$showDialog$0(DockingErrorDisplay.java:91)
			at ghidra.util.Swing.runIfSwingOrRunLater(Swing.java:124)
			at ghidra.util.SystemUtilities.runIfSwingOrPostSwingLater(SystemUtilities.java:239)
			at docking.DockingErrorDisplay.showDialog(DockingErrorDisplay.java:90)
			at docking.DockingErrorDisplay.displayMessage(DockingErrorDisplay.java:78)
			at docking.DockingErrorDisplay.displayErrorMessage(DockingErrorDisplay.java:42)
			at ghidra.util.Msg.showError(Msg.java:333)
			at ghidra.util.task.SwingUpdateManager.doWork(SwingUpdateManager.java:324)
			at ghidra.util.task.SwingUpdateManager.checkForWork(SwingUpdateManager.java:260)
			at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
			at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
			at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
			at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
			at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
			at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
			at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
			at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
			at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
			at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
			at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
			at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
			at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

**Environment (please complete the following information):**
 - OS: ArchLinux x64
 - Java Version: OpenJDK 13
 - Ghidra Version: bcb825fb029232175625bc85653ec0e810b1252e

**Comment by saruman9** [comment]
I can't reproduce the error. My environment:
- OS: ArchLinux;
- Java Version: OpenJDK 13 (also tested with 11);
- Ghidra Version: bcb825f.

My target was `libstdc++.so.6.0.28` from Arch-based distributions.

May be you use external analyzers (extensions). Are you tried to reproduce the error on clear installed Ghidra?

**Comment by astrelsky** [comment]
> I can't reproduce the error. My environment:
> - OS: ArchLinux;
> - Java Version: OpenJDK 13 (also tested with 11);
> - Ghidra Version: bcb825f.
> 
> My target was `libstdc++.so.6.0.28` from Arch-based distributions.
> 
> May be you use external analyzers (extensions). Are you tried to reproduce the error on clear installed Ghidra?

I have had this occur. It's not reproducible. Of course when it does occur I'm not running in a debugger :/

**Comment by dragonmacher** [comment]
I can see how this could happen.  We should be able to protect against the exception.

**Comment by dragonmacher** [comment]
Fixed by 11e187b

**Comment by saruman9** [comment]
> Fixed by 11e187b

Sorry, I can't find a commit with the hash (and GitHub don't create link). Can you post the link to commit?

**Comment by astrelsky** [comment]
> > Fixed by 11e187b
> 
> Sorry, I can't find a commit with the hash (and GitHub don't create link). Can you post the link to commit?

It probably hasn't been mirrored yet.

**Comment by dragonmacher** [comment]
> > > Fixed by 11e187b
> > 
> > 
> > Sorry, I can't find a commit with the hash (and GitHub don't create link). Can you post the link to commit?
> 
> It probably hasn't been mirrored yet.

Correct
